[mod] engines - add Onesearch engine

return42 opened the pull request:
## What does this PR do?

Add Onesearch engine / Merged from @e-foundation's onesearch-engine branch [1] and slightly modfied / see [2].

[1] https://github.com/e-foundation/searx/tree/onesearch-engine
[2] https://github.com/searx/searx/pull/3065

**pro**: engine does fast response
~**hint**: engine is enabled by default~
**hint**: is literally just **Bing** --> could be a replacement for the bing engine / that is enabled by default but has some issues

**ToDo:** :

- [ ] Engine needs some polish .. this version does not support languages
- [ ] images: https://images.onesearch.com/yhs/search;?p=paris
- [ ] videos: https://video.onesearch.com/yhs/search;?p=paris
- [ ] news news: https://images.onesearch.com/yhs/search;?p=covid

-----
OneSearch is literally just **Bing** results flanked by ads that don’t track you despite being from a company that makes money tracking you [ref(https://lifehacker.com/is-yahoos-new-onesearch-engine-good-for-privacy-1841042875)].

According to the OneSearch privacy policy, search results will only be personalized based on location, which it will collect from IP addresses. OneSearch says that it will separate IP addresses from users and their search results [ref](https://www.theverge.com/2020/1/14/21065640/verizon-onesearch-privacy-tracking-yahoo-breach-hack).

## How to test this PR locally?

    !one search term

israelyago committed:
[mod] engines - add Onesearch engine

Merged from @e-foundation's onesearch-engine branch [1] and slightly modfied /
see [2].

[1] https://github.com/e-foundation/searx/tree/onesearch-engine
[2] https://github.com/searx/searx/pull/3065

Autor: Israel Yago Pereira <israelyago@e.email>
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 commented on searx/data/engine_descriptions.json:
```diff
@@ -992,6 +1058,10 @@
+  ],
+  "onesearch":[
+   "OneSearch is a free privacy-oriented search experience that lets you explore what you care about without unwanted profile or ad targeting. Take back control of your data today!",
+   "https://www.onesearch.com"
```
The script `./searxng_extra/update/update_engine_descriptions.py` has found a documentation / even there exists no wikidata entry.

Compare with https://github.com/searxng/searxng/pull/513/files#r751783403 / no words about Yahoo & Verizon, companies that are not known for protecting privacy.

return42 commented on searx/engines/onesearch.py:
```diff
@@ -0,0 +1,67 @@
+
+- https://www.onesearch.com
+
+OneSearch is literally just Bing results flanked by ads that don’t track you
+despite being from a company that makes money tracking you [1].
+
+According to the OneSearch privacy policy, search results will only be
+personalized based on location, which it will collect from IP addresses.
+OneSearch says that it will separate IP addresses from users and their search
+results [2].
+
+[1] https://lifehacker.com/is-yahoos-new-onesearch-engine-good-for-privacy-1841042875
+[2] https://www.theverge.com/2020/1/14/21065640/verizon-onesearch-privacy-tracking-yahoo-breach-hack
```
compare with https://github.com/searxng/searxng/pull/513/files#r751782506

dalf replied:
I've added a wikidata entry: https://www.wikidata.org/wiki/Q109682354 so `update_engine_descriptions.py` should the wikidata description.

return42 replied:
> I've added a wikidata entry

Very cool .. 

https://github.com/searxng/searxng/blob/c4a8c77026319185f516567f9b288157c2f87e20/searx/engines/onesearch.py#L28-L30

and I can confirm update_engine_descriptions.py works:

https://github.com/searxng/searxng/blob/356cff0692e0c59d263c17964dfb3a45ee3b2305/searx/data/engine_descriptions.json#L1053-L1056

https://github.com/searxng/searxng/blob/356cff0692e0c59d263c17964dfb3a45ee3b2305/searx/data/engine_descriptions.json#L1734-L1737

![grafik](https://user-images.githubusercontent.com/554536/143865625-f33c4ef8-e5e7-4e42-9d25-3d63c68f810b.png)

return42 committed:
[upd] engine_descriptions.json

Updated by:

    ./manage pyenv.cmd python3 \
        ./searxng_extra/update/update_engine_descriptions.py

Include documentation of new added engines:

- OneSearch (Yahoo & Verizon)
- superuser
- stackoverflow
- askubuntu
- indb
- ...

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
WIP
